File: Server/src/main/java/bgu/spl171/net/impl/TFTP/Commands/Broadcast.java
package bgu.spl171.net.impl.TFTP.Commands;

import bgu.spl171.net.impl.TFTP.Commands.Responses.Response;

public class Broadcast implements Command {
    boolean isAdded;
    String fileName;

    public Broadcast(boolean isAdded, String fileName) {
        this.isAdded = isAdded;
        this.fileName = fileName;
    }

    public boolean isAdded() {
        return isAdded;
    }

    public String getFileName() {
        return fileName;
    }

    @Override
    public short getOpCode() {
        return OpCode.Broadcast.getValue();
    }

}

File: Client/src/Commands/Broadcast.cpp
#include "../../include/Commands/Broadcast.h"

Broadcast::Broadcast(bool isAdded, const std::string &fileName) {
    this->isAdded_Renamed = isAdded;
    this->fileName = fileName;
}

bool Broadcast::isAdded() {
    return isAdded_Renamed;
}

std::string Broadcast::getFileName() {
    return fileName;
}

short Broadcast::getOpCode() {
    return opcode::Broadcast;
}
